src/responder/component.ts

```diff
@@ -0,0 +1,182 @@
+import { PisaTransactionIdentifier } from "./gasQueue";
+import {
+    MappedState,
+    StateReducer,
+    MappedStateReducer,
+    Component,
+    BlockNumberState,
+    BlockNumberReducer
+} from "../blockMonitor/component";
+import { ReadOnlyBlockCache } from "../blockMonitor";
+import { Block } from "../dataEntities";
+import { MultiResponder } from "./multiResponder";
+import { ResponderBlock } from "../dataEntities/block";
+
+export enum ResponderStateKind {
+    Pending = 1,
+    Mined = 2
+}
+export type PendingResponseState = {
+    appointmentId: string;
+    kind: ResponderStateKind.Pending;
+    identifier: PisaTransactionIdentifier;
+};
+export type MinedResponseState = {
+    appointmentId: string;
+    kind: ResponderStateKind.Mined;
+    identifier: PisaTransactionIdentifier;
+    blockMined: number;
+    nonce: number;
+};
+export type ResponderAppointmentAnchorState = PendingResponseState | MinedResponseState;
+export type ResponderAnchorState = MappedState<ResponderAppointmentAnchorState> & BlockNumberState;
+
+/**
+ * Selects information from blocks that are relevant to generating responses.
+ */
+export class ResponderAppointmentReducer implements StateReducer<ResponderAppointmentAnchorState, ResponderBlock> {
+    public constructor(
+        private readonly blockCache: ReadOnlyBlockCache<ResponderBlock>,
+        private readonly identifier: PisaTransactionIdentifier,
+        private readonly appointmentId: string,
+        private readonly address: string
+    ) {}
+
+    private txIdentifierInBlock(
+        block: ResponderBlock,
+        identifier: PisaTransactionIdentifier
+    ): { blockNumber: number; nonce: number } | null {
+        for (const tx of block.transactions) {
+            // a contract creation - cant be of interest
+            if (!tx.to) continue;
+
+            // look for matching transactions
+            const txIdentifier = new PisaTransactionIdentifier(tx.chainId, tx.data, tx.to, tx.value, tx.gasLimit);
+            if (txIdentifier.equals(identifier) && tx.from.toLocaleLowerCase() === this.address.toLocaleLowerCase()) {
```

Comment:
I think toLowerCase would be better here, as the locale doesn't matter.

Reply:
Yep, locale was a mistake.